### ISSUE TITLE
Optimize UTs

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#276]
Something that I like to work is on the optimization of UTs. There is no much problems on it currently, but I like to normalize UTs to improve readability and testability. One problem that I don't know if is with me only is about QFTs, seems that when it fails it just report the fail, not the actual vs expected result, and I can fix that.

I have done some internal changes, and I reach to a result like that:

```java
public void testInspections() {
    checkInspections("fixtures/api/empty-function.php", 
        IsEmptyFunctionUsageInspection.class);
}

public void testQuickFixes() {
    checkAllQuickFixes("fixtures/api/empty-function.php", 
        IsEmptyFunctionUsageInspection.class);
}
```

It should be done in `*Test.java` files from project. I still need check if it will works for all cases before starts, but for now, I don't see any problem.

**Comment by funivan** [comment]
I use similar approach in other branch (https://github.com/funivan/phpinspectionsea/blob/222-naming-convention/src/test/java/com/kalessil/phpStorm/phpInspectionsEA/codeStyle/NamingConventionInspectorTest.java#L35:L45) 
Agree with you. 

p.s. I event don't know what mean this code `myFixture.testHighlighting(true, false, true);`)))

**Comment by funivan** [comment]
We should add label `internal`

**Comment by rentalhost** [comment]
The `myFixture.testHighlighting()` will get the pure PHP code from tested file (it is, without the XML code), then will run all inspections (that will regenerate the XML) then will compare if "original XML+PHP" is equal to "generated XML+PHP".

I agree about `internal` label.

**Comment by kalessil** [comment]
May I cut in? Do we really need to touch tests (it's even not all inspections covered, I'll keep in mind name new once properly ofcourse)?

**Comment by rentalhost** [comment]
I want do that for two reasons: it is the first step. The unit tests are not the major problem currently, we have to create UTs for all inspections and too QFTs. But if we normalize that, make it more simple, we can improve the testability.

As I mentined, I don't know if is only with me, but on run some QFTs I don't see the "click to see difference". But it is implementable (I have did internally). And it is a blocker to start the creation of QFs and QFTs, for instance, because we can't know what was the result of implemented QF.

The second step (_but I will open after close that_) is about code coverage. The focus is reduce duplicated tests (or that have "the same meaning") and check if all code is catched (in a simple run, I found some dead codes/untested, and I belive that it could be a problem).

I know that tests are "internal things", but it affects indirectly the end user. If it fails, user will see the fail in some moment.

**Comment by funivan** [comment]
I have only one problem with UT. When I create unit test in intellij i should change location of this test. I would rather prefer to synchronize location of test cases.

**Comment by kalessil** [comment]
@funivan @rentalhost Sold, go ahead. Feel free to share work and re-work the structure on your taste.

**Comment by kalessil** [comment]
Closing, too risky for breaking merge into the Ultimate codebase.
